### PR TITLE
Add car

### PR DESCRIPTION
**Pull request by alimalone**

### CODE REVIEW

**Review by laurentmahieu — APPROVED** [review]

**Review by laurentmahieu — DISMISSED** [review]

File: backend/route/modelevehicule.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const express = require("express");
+const { connection } = require("../conf");
+const router = express.Router();
+
+//route du modèle de voiture
+router.get("/:marque", (req, res) => {
+  const modele = req.params.id;
+  connection.query(
+    `SELECT *
```
Comment (BlandineM) [inline backend/route/modelevehicule.js]: no *

**Review by BlandineM — DISMISSED** [review]

**Review by BlandineM — DISMISSED** [review]

**Review by laurentmahieu — DISMISSED** [review]

**Review by laurentmahieu — APPROVED** [review]

**Review by laurentmahieu — DISMISSED** [review]

File: frontend/src/pages/SelectCar.jsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    axios.get(`${apiSite}/modelevehicule/marque`).then(({ data }) => {
+      initModeles(data);
+      limitStringLength();
+    });
+  }, []);
+
+  function addCar(e) {
+    e.preventDefault();
+    if (vin.length !== 17) {
+      //insert toast
```
Comment (LoicBrassart) [inline frontend/src/pages/SelectCar.jsx]: Do not leave "TODO" comments

File: frontend/src/pages/SelectCar.jsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    } else if (parseInt(kilometrage) >= Math.pow(10, 6) || isNaN(kilometrage)) {
+      //insert toast
```
Comment (LoicBrassart) [inline frontend/src/pages/SelectCar.jsx]: Do not leave "TODO" comments

File: frontend/src/pages/SelectCar.jsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    } else if (
+      parseInt(année) < 1900 ||
+      parseInt(année) > year ||
+      isNaN(année)
+    ) {
+      //insert toast
```
Comment (LoicBrassart) [inline frontend/src/pages/SelectCar.jsx]: Do not leave "TODO" comments

File: frontend/src/pages/SelectCar.jsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    } else {
+      //insert toast
```
Comment (LoicBrassart) [inline frontend/src/pages/SelectCar.jsx]: Do not leave "TODO" comments

File: frontend/src/pages/SelectCar.jsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    }
+  }
+
+  function limitStringLength() {
```
Comment (LoicBrassart) [inline frontend/src/pages/SelectCar.jsx]: I don't get this function, what is it supposed to do ?

File: frontend/src/pages/SelectCar.jsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    return (
+      document.getElementById("année").maxLength,
+      document.getElementById("kilometrage").maxLength,
+      document.getElementById("VIN").maxLength
+    );
+  }
+
+  function getData() {
```
Comment (LoicBrassart) [inline frontend/src/pages/SelectCar.jsx]: This function is never called, anywhere

File: frontend/src/pages/Signup.jsx [hunk 8]
```diff
@@ -21,6 +21,7 @@ function Signup(props) {
   };
 
   function signup(e) {
+
```
Comment (LoicBrassart) [inline frontend/src/pages/Signup.jsx]: This change is useless

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Some changes to do

File: frontend/src/pages/SelectCar.jsx (hunk 2 above)
Comment (laurentmahieu) [reply frontend/src/pages/SelectCar.jsx]: done

File: frontend/src/pages/SelectCar.jsx (hunk 3 above)
Comment (laurentmahieu) [reply frontend/src/pages/SelectCar.jsx]: done

File: frontend/src/pages/SelectCar.jsx (hunk 4 above)
Comment (laurentmahieu) [reply frontend/src/pages/SelectCar.jsx]: done

File: frontend/src/pages/SelectCar.jsx (hunk 5 above)
Comment (laurentmahieu) [reply frontend/src/pages/SelectCar.jsx]: done

File: frontend/src/pages/SelectCar.jsx (hunk 6 above)
Comment (laurentmahieu) [reply frontend/src/pages/SelectCar.jsx]: this function (" limitStringLength ") is to limit the number of entry string in the input text by their restriction value.
ex: "année" has maxLength="4" so we can't entry a number of 5 strings in the input.
"Vin" has maxLength="17"
"km" has maxLength="7"

File: frontend/src/pages/SelectCar.jsx (hunk 7 above)
Comment (laurentmahieu) [reply frontend/src/pages/SelectCar.jsx]: deleted

File: frontend/src/pages/Signup.jsx (hunk 8 above)
Comment (laurentmahieu) [reply frontend/src/pages/Signup.jsx]: exactly ^^

**Review by BlandineM — DISMISSED** [review]

**Review by laurentmahieu — DISMISSED** [review]

**Review by laurentmahieu — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
.

**Pull request merged**